test/e2e/k8s_util.go

```diff
@@ -700,56 +701,85 @@ func (k *KubernetesUtils) waitForPodInNamespace(ns string, pod string) ([]string
 func (k *KubernetesUtils) waitForHTTPServers(allPods []Pod) error {
 	const maxTries = 10
 	log.Infof("waiting for HTTP servers (ports 80, 81 and 8080:8085) to become ready")
-	log.Infof("Waiting for HTTP servers (ports 80, 81, 5000 and 8080:8085) to become ready")
-	var wrong int
-	for i := 0; i < maxTries; i++ {
+
+	serversAreReady := func() bool {
 		reachability := NewReachability(allPods, Connected)
-		k.Validate(allPods, reachability, 80, v1.ProtocolTCP)
-		k.Validate(allPods, reachability, 80, v1.ProtocolUDP)
-		k.Validate(allPods, reachability, 80, v1.ProtocolSCTP)
-		k.Validate(allPods, reachability, 81, v1.ProtocolTCP)
-		k.Validate(allPods, reachability, 81, v1.ProtocolUDP)
-		k.Validate(allPods, reachability, 81, v1.ProtocolSCTP)
-		for j := 8080; j < 8086; j++ {
-			k.Validate(allPods, reachability, int32(j), v1.ProtocolTCP)
+		var wrong int
+		k.Validate(allPods, reachability, []int32{80, 81, 8080, 8081, 8082, 8083, 8084, 8085}, v1.ProtocolTCP)
+		_, wrong, _ = reachability.Summary()
+		if wrong != 0 {
+			return false
```

Comment:
nit
```suggestion
		if _, wrong, _ := reachability.Summary(); wrong != 0 {
		        return false
		}
```

Reply:
Will address with other comments, if there is any.

Reply:
Done.